code/a/analyze_nl_boot.do

```diff
@@ -0,0 +1,138 @@
+/****************************************************************/
+/* XC REGRESSIONS FROM analyze_nl.do WITH CONSUMPTION BOOTSTRAP */
+/****************************************************************/
+
+/*****************************/
+/* CROSS-SECTION REGRESSIONS */
+/*****************************/
+
+/* bivariate correlation tables.
+ Column 1. District (no FE)
+ Column 2. Subdistrict (no FE)
+ Column 3. Town (within district)
+ Column 4. shrid (within district)
+ Column 5. shrid (within subdistrict)
+*/
+
+/* set number of bootstraps */
+global nboot 1000
+
+/* keep full variable list since it affects what we will drop */
+global tvars ln_pc11_pca_tot_p ln_ec13_emp_all ln_ec13_emp_manuf ln_ec13_emp_serv ln_consumption pc11_td_power_share
+global vvars ln_pc11_pca_tot_p ln_ec13_emp_all ln_ec13_emp_manuf ln_ec13_emp_serv ln_consumption ln_pc11_power 
+
+/* prepare the estimates file */
+global tmpfile $tmp/nlxc_ests.csv
+cap erase $tmpfile
+append_to_file using $tmpfile, s(b,se,p,n,var,boot)
+
+/*********************************************/
+/* district and subdistrict-level bootstraps */
+/*********************************************/
+di "Bootstrapping dist / subdistricts..."
+qui forval b = 1/$nboot {
+  noi di `b'
+  use $tmp/shrugboot/shrug_cons_district_both_boot_`b'.dta, clear
+  quireg ln_cons ln_light2012 ln_num_cells, robust 
+  append_est_to_file using $tmpfile, b(ln_light2012) s(dxc_cons,`b') 
+
+  use $tmp/shrugboot/shrug_cons_subdistrict_both_boot_`b'.dta, clear
+  quireg ln_cons ln_light2012 ln_num_cells, robust 
+  append_est_to_file using $tmpfile, b(ln_light2012) s(sdxc_cons,`b') 
+}
+
+/********************/
+/* town-level stats */
+/********************/
+
+/* open shrid data */
+use $tmp/shrug_combined, clear
+group pc11_state_id
+drop if pc11_sector == 2
+foreach v in $tvars {
+  drop if mi(`v')
+}
+
+/* keep only the vars we use for the regs */
+keep shrid ln_light2012 ln_num_cells sdgroup ln_consumption
+
+/* replace consumption data with bootstrap data B */
+merge 1:1 shrid using $secc/parsed_draft/dta/ancillary/shrug_urban_cons_boot, keepusing(secc_cons_pc_*) keep(match) nogen
+
+/* loop over bootstraps */
+di "Bootstrapping towns..."
+qui forval b = 1/$nboot {
+
+  noi di `b'
+  
+  /* set consumption to the current bootstrap value */
+  replace ln_consumption = ln(secc_cons_pc_`b')
+  
+  /* run the regressions and store the bootstrap coef */
+  quireg ln_consumption ln_light2012 ln_num_cells, robust title(ln_consumption) absorb(sdgroup)
+  append_est_to_file using $tmpfile, b(ln_light2012) s(txc_d_cons,`b') 
+
```

Comment:
delete this blank line

Reply:
fixed in this [commit](https://github.com/devdatalab/paper-shrug/commit/ac6b4450ab5172fc6e95e16118c4a0b0ffefd180)